### DISCUSSION TITLE
UI refactoring

### DISCUSSION THREAD

**Discussion by exquo** [isamert/scli#110] [Show and tell]
I'm doing some refactoring in [ui-misc](https://github.com/isamert/scli/tree/ui-misc-WIP) branch. It's pretty much done, I'm just polishing the `pylint`'s complaints, and will split it into logical commits. It would certainly be easier to read smaller commits (with actual comments!:) rather then a single 2k lines diff, but just in case you want to do a test drive - I'm all ears for the feedback!

**Comment by isamert** [comment]
Ouch, I totally missed this branch. I was making a few changes (generally quite small) but I guess I should wait for your changes to get merged. The diff is quite unreadable right now but I feel like it's making things better :)

I just found this issue: hitting `:` and `/` outside the input box crashes scli. Other than that everything seems smooth after my 5-min ride.

**Comment by exquo** [comment]
Thanks! Just a heads up on what I'm up to :)

This shouldn't affect your development at all! You don't have to wait - I might get sidetracked with something else, and the branch might sit there unmerged for a month :). On the other hand, if I commit it before you submit your modifications - no problem either, I can merge your changes, so that you don't have to figure out where the pieces go in the refactored version.

**Comment by exquo** [comment]
> hitting : and / outside the input box crashes scli

Hmm, I don't seem to get that.. Tried removing `sclirc` to simulate a 'pure' setup but still - no crash.

**Comment by exquo** [comment]
Now fully merged in `master`.
